app/src/main/java/com/github/epfl/meili/photo/PhotoEditActivity.kt

```diff
@@ -57,19 +59,24 @@ class PhotoEditActivity : AppCompatActivity() {
         binding.sharpen.setOnClickListener { photoEditor.setFilterEffect(PhotoFilter.SHARPEN) }
         binding.fishEye.setOnClickListener { photoEditor.setFilterEffect(PhotoFilter.FISH_EYE) }
         binding.saturate.setOnClickListener { photoEditor.setFilterEffect(PhotoFilter.SATURATE) }
-
-
-        if (!isPermissionGranted()) {
-            getStoragePermission() // photo will get stored on phone once it is done being edited
-        }
+        // text
+        binding.tvText.setOnClickListener { toggleText() }
+        //enable text editing
+        photoEditor.setOnPhotoEditorListener( object: OnPhotoEditorListener { override fun onEditTextChangeListener(rootView: View, text: String, colorCode: Int) { photoEditor.editText(rootView, binding.etText.text.toString(), binding.colorSlider.color); }
+            override fun onAddViewListener(p0: ViewType?, p1: Int) {} override fun onRemoveViewListener(p0: ViewType?, p1: Int) {}
+            override fun onStartViewChangeListener(p0: ViewType?) {} override fun onStopViewChangeListener(p0: ViewType?) {} })
+
+        // photo will get stored on phone once it is done being edited
+        if (!isPermissionGranted()) { getStoragePermission() }
         setFabListener()
 
         // Handle emojis
-        binding.emojis.setOnClickListener { showEmojiTable() }
+        /** Show scrollable emoji table */
```

Comment:
Nitpick : this comment is a bit weird now that it's not above a function anymore. Maybe you could delete it or make it // ?